lib/transform/resolveNestedDefinitionTransformer.ts

```diff
@@ -11,10 +11,17 @@ export const resolveNestedDefinitionTransformer: SpecTransformer = {
   transform(spec, { jsonLoader, objSchemas, arrSchemas, primSchemas, allParams }) {
     const queue = new Array<string>();
 
-    const visitNestedDefinitions = (s: Schema | undefined, ref?: string) => {
+    const visitNestedDefinitions = (
+      s: Schema | undefined,
+      ref?: string,
+      isResponseSchema?: boolean
+    ) => {
       if (s === undefined || s === null || typeof s !== "object") {
         return;
       }
+      if (isResponseSchema && s.type === "object" && s.properties === undefined) {
+        s.noRefWithTypeObject = true;
```

Comment:
Can you rename `noRefWithTypeObject`? This is confuse as it validates schema doesn't have properties defined rather than no $ref.